recipe/patches-grpc/0001-Link-libraries-privately.patch

```diff
@@ -0,0 +1,61 @@
+From ab7a219a7267a7f773286ba5f6e3590cb7d609a5 Mon Sep 17 00:00:00 2001
+From: Mark Harfouche <mark.harfouche@gmail.com>
+Date: Mon, 15 Aug 2022 15:24:27 -0400
+Subject: [PATCH] Link libraries privately
+
+---
+ CMakeLists.txt | 36 ++++++++++++++++++------------------
+ 1 file changed, 18 insertions(+), 18 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 2bfbcd4c6b..cb2ab07f16 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -2137,26 +2137,26 @@ target_include_directories(grpc
+     ${_gRPC_ZLIB_INCLUDE_DIR}
+ )
+ target_link_libraries(grpc
+-  ${_gRPC_BASELIB_LIBRARIES}
+-  ${_gRPC_ZLIB_LIBRARIES}
+-  ${_gRPC_CARES_LIBRARIES}
+-  ${_gRPC_ADDRESS_SORTING_LIBRARIES}
+-  ${_gRPC_RE2_LIBRARIES}
+-  ${_gRPC_UPB_LIBRARIES}
+-  ${_gRPC_ALLTARGETS_LIBRARIES}
+-  absl::flat_hash_map
+-  absl::inlined_vector
+-  absl::bind_front
+-  absl::hash
+-  absl::statusor
+-  absl::variant
+-  absl::utility
+-  gpr
+-  ${_gRPC_SSL_LIBRARIES}
+-  address_sorting
++  PRIVATE ${_gRPC_BASELIB_LIBRARIES}
++  PRIVATE ${_gRPC_ZLIB_LIBRARIES}
++  PRIVATE ${_gRPC_CARES_LIBRARIES}
++  PRIVATE ${_gRPC_ADDRESS_SORTING_LIBRARIES}
++  PRIVATE ${_gRPC_RE2_LIBRARIES}
++  PRIVATE ${_gRPC_UPB_LIBRARIES}
++  PRIVATE ${_gRPC_ALLTARGETS_LIBRARIES}
++  PRIVATE absl::flat_hash_map
++  PRIVATE absl::inlined_vector
++  PRIVATE absl::bind_front
++  PRIVATE absl::hash
++  PRIVATE absl::statusor
++  PRIVATE absl::variant
++  PRIVATE absl::utility
++  PRIVATE gpr
++  PRIVATE ${_gRPC_SSL_LIBRARIES}
++  PRIVATE address_sorting
+ )
+ if(_gRPC_PLATFORM_IOS OR _gRPC_PLATFORM_MAC)
+-  target_link_libraries(grpc "-framework CoreFoundation")
++  target_link_libraries(grpc PRIVATE "-framework CoreFoundation")
+ endif()
```

Comment:
Something ate my comment here. The issue was that calling `target_link_libraries` again then needs to set a PRIVATE/PUBLIC/INTERFACE here, then configuration succeeds. I guessed private based on some SO questions, but don't really know TBH.

Reply:
```suggestion
 if(_gRPC_PLATFORM_IOS OR _gRPC_PLATFORM_MAC)
-  target_link_libraries(grpc "-framework CoreFoundation")
+  target_link_libraries(grpc PUBLIC "-framework CoreFoundation")
 endif()
```